packages/@aws-cdk/cdk/lib/app.ts

```diff
@@ -125,27 +188,25 @@ export class App extends Root {
     }
   }
 
-  private collectRuntimeInformation(): cxapi.AppRuntime {
-    const libraries: { [name: string]: string } = {};
+  protected prepare() {
+    this.prepared = true;
+  }
 
-    for (const fileName of Object.keys(require.cache)) {
-      const pkg = findNpmPackage(fileName);
-      if (pkg && !pkg.private) {
-        libraries[pkg.name] = pkg.version;
-      }
+  protected validate(): string[] {
+    this.validated = true;
+    if (numberOfAppsInProgram(this) > 1 && this.node.id === DEFAULT_APP_NAME) {
```

Comment:
But what if the `id` I passed is `App` (which happens to be the default)? Shouldn't this refer to the `defaultAppName` property?